packages/ai-chat/src/common/chat-service.ts

```diff
@@ -255,6 +260,28 @@ export class ChatServiceImpl implements ChatService {
         return invocation;
     }
 
+    protected updateSessionMetadata(session: ChatSessionInternal, request: MutableChatRequestModel): void {
+        session.lastInteraction = new Date();
+        if (session.title) {
+            return;
+        }
+        const requestText = request.request.displayText ?? request.request.text;
+        session.title = requestText;
+        if (this.chatSessionNamingService) {
+            const otherSessionNames = this._sessions.map(s => s.title).filter((title): title is string => title !== undefined);
+            const namingService = this.chatSessionNamingService;
+            let didGenerateName = false;
+            request.response.onDidChange(() => {
+                if (request.response.isComplete && !didGenerateName) {
+                    namingService.generateChatSessionName(session, otherSessionNames).then(name => {
+                        session.title = name;
+                        didGenerateName = true;
+                    }).catch(error => this.logger.error('Failed to generate chat session name', error));
+                }
+            });
```

Comment:
This does not take into account when the user manually edited the name while the response was not complete, i.e. the user name will be overwritten.